### PR TITLE
ENG-26 Implement anomaly likelihood processing in unicorn backend.

### PR DESCRIPTION
**Pull request by vitaly-krugl**
@scottpurdy, please review _ModelRunner class changes to make sure it uses NuPIC's anomaly likelihood algorithm correctly.

IMPORTANT: this PR depends on a new release of nupic (0.2.12?) that will include the NuPIC PR https://github.com/numenta/nupic/pull/2535

### CODE REVIEW

**Comment by vitaly-krugl** [comment]
NOTE: Merge of this PR is blocked by https://github.com/numenta/nupic/pull/2544 and release of nupic 0.2.12

**Comment by vitaly-krugl** [comment]
The relevant nupic functionality (new AnomalyLikelihood class features) has now been released both internally and externally as nupic v 0.3.0, so merging this PR...

CC @brev, @lscheinkman, @marionleborgne

File: unicorn/tests/py/integration/model_runner_test.py [hunk 1]
```diff
@@ -181,7 +181,7 @@ def testFeedInputGetOutput(self):
         rowIndex, anomalyLikelihood = json.loads(outputRecord)
 
         self.assertEqual(rowIndex, i)
-        self.assertIsInstance(anomalyLikelihood, float)
+        self.assertIsInstance(anomalyLikelihood, (float, int))
```
Comment (scottpurdy) [inline unicorn/tests/py/integration/model_runner_test.py]: in what case will it be an `int`?

**Comment by scottpurdy** [comment]
:+1:

File: unicorn/tests/py/integration/model_runner_test.py (hunk 1 above)
Comment (vitaly-krugl) [inline unicorn/tests/py/integration/model_runner_test.py]: @scottpurdy: Good catch. When I first started, unaware of the AnomalyLikelihood class in NuPIC (also before my latest changes to the AnomalyLikelihood class), I had my own implementation of the class based on the AnomalyLikelihoodHelper that returned 0 (an int) during the probationary period. Now that we're using the AnomalyLikelihood class that returns 0.5 (float) during the probationary period, this `(float, int)` change is technically unnecessary.

Comment (vitaly-krugl) [inline unicorn/tests/py/integration/model_runner_test.py]: @scottpurdy: see #325

**Pull request merged**